src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -371,11 +371,22 @@ void drawing_classifier::init_training(
   // by the data iterator.
   nn_spec_ = init_model();
 
-  if ( enable_warmstart ) {
+  if (mlmodel_path.size()) {
     // Initialize the neural net with warm start model weights.
     model_spec warmstart_model(mlmodel_path);
     float_array_map trained_weights = warmstart_model.export_params_view();
     nn_spec_->update_params(trained_weights);
+
+    size_t idx = 0;
+#ifdef _WIN32
+    idx = mlmodel_path.rfind('\\');
+#else
+    idx = mlmodel_path.rfind('/');
+#endif
+    if (idx == std::string::npos)
+      state["warm_start"] = mlmodel_path;
+    else
+      state["warm_start"] = mlmodel_path.substr(idx + 1);
```

Comment:
Isn't this option populated by Python? Let's avoid duplicating the logic here. This state should reflect what the user selected, not the path that is the implementation detail

Reply:
Don't you have to store the `option` into `state`?

Reply:
Here, I'm storing the model name into the `state`. Not manipulating the option. The `state` will be serialized into disk when we save model.

Reply:
I believe the option gets stored to the state by https://github.com/apple/turicreate/blob/master/src/toolkits/drawing_classifier/drawing_classifier.cpp#L248 .

Reply:
ha, I confused [model_path](https://github.com/apple/turicreate/blob/58527ce50121455b1e201198c1a4fbe2e695cf14/src/toolkits/drawing_classifier/drawing_classifier.cpp#L311) with `wram_start`. good catch.